packages/salesforcedx-sobjects-faux-generator/src/generator/fauxClassGenerator.ts

```diff
@@ -1,33 +1,245 @@
 /*
- * Copyright (c) 2017, salesforce.com, inc.
- * All rights reserved.
- * Licensed under the BSD 3-Clause license.
- * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
- */
+* Copyright (c) 2017, salesforce.com, inc.
+* All rights reserved.
+* Licensed under the BSD 3-Clause license.
+* For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+*/
 import * as fs from 'fs';
 import * as path from 'path';
-import { SObjectCategory, SObjectDescribe } from '../describe';
+
+import { EOL } from 'os';
+import { mkdir, rm } from 'shelljs';
+
+import {
+  ChildRelationship,
+  Field,
+  SObject,
+  SObjectCategory,
+  SObjectDescribe
+} from '../describe';
 
 export class FauxClassGenerator {
-  public async generate(projectPath: string, type: SObjectCategory) {
+  private SFDX_DIR = '.sfdx';
+  private TOOLS_DIR = 'tools';
+  private SOBJECTS_DIR = 'sobjects';
+  private STANDARDOBJECTS_DIR = 'standardObjects';
+  private CUSTOMOBJECTS_DIR = 'customObjects';
+
+  public async generate(
+    projectPath: string,
+    type: SObjectCategory
+  ): Promise<string> {
+    const sobjectsFolderPath = path.join(
+      projectPath,
+      this.SFDX_DIR,
+      this.TOOLS_DIR,
+      this.SOBJECTS_DIR
+    );
+
+    this.cleanupSObjectFolders(sobjectsFolderPath);
+
     const describe = new SObjectDescribe();
     const sobjects = await describe.describeGlobal(projectPath, type);
-    console.log(sobjects.length);
-    for (let i = 0; i < sobjects.length; i++) {
-      const sobject = await describe.describeSObject(projectPath, sobjects[i]);
-      console.log(sobject.name);
-      this.generateFauxClass(projectPath, sobject);
+    const standardSObjects: SObject[] = [];
+    const customSObjects: SObject[] = [];
+    let fetchedSObjects: SObject[] = [];
+    let j = 0;
+    while (j < sobjects.length) {
+      try {
+        fetchedSObjects = fetchedSObjects.concat(
+          await describe.describeSObjectBatch(projectPath, sobjects, j)
```

Comment:
This might be a long-hanging fruit. We might be doing things too sequentially here.
See the section "Careful! Avoid going too sequential" at https://developers.google.com/web/fundamentals/getting-started/primers/async-functions

Doesn't need to be fixed in this PR but I want  you to be aware of it.

CC: @chowwinston @DatGuyJonathan @guw @Sweetman

Reply:
interesting - I can try doing the batch fetches in parallel and see if that helps - good spot